#7 -HibpClient

bojanrmv opened the pull request:
- Created HttpClientFactory for creating HttpClient with base api address
- internal constructor of HIBPClient for DI uses.
- public constructor for non DI uses accepting api key
#7

bojanrmv committed:
- Created HttpClientFactory for creating HttpClient with base api address
- internal constructor of HIBPClient for DI uses.
- public constructor for non DI uses accepting api key

bojanrmv committed:
- Created HttpClientFactory for creating HttpClient with base api address
- internal constructor of HIBPClient for DI uses.
- public constructor for non DI uses accepting api key

danpdc commented:
@zvaklinov  and @bojanrmv I think you two need to synchronize a little bit to understand what constants to keep and commit only one version of the constants. So one of you will remove it from the PR. 

Also, @zvaklinov make sure that you review the comments and we should merge your PR first. 
Then @bojanrmv you should to a rebase of your local branch and you should be able to work with the new HibpCLientOptions class.

zvaklinov committed:
Implement the HibpClientOptions class (#6) (#14)

* Implement the HibpClientOptions class (#6)

- Add the base options for using the API;
- Add a constant for the BaseAddress (base URL for the HIBP API).

* Add XML comments to describe the HibpClientOptions class (#6)

---------

Co-authored-by: Zahari Vaklinov <zahari.vaklinov@amdaris.com>

bojanrmv committed:
HttpClient Factory with HIBP options

bojanrmv committed:
typo

bojanrmv committed:
added missing namespace
